src/applicationcomposer/activation.ts

```diff
@@ -4,12 +4,41 @@
  */
 
 import * as vscode from 'vscode'
+import * as CloudFormation from '../shared/cloudformation/cloudformation'
 import { ApplicationComposerManager } from './webviewManager'
 import { Commands } from '../shared/vscode/commands2'
 import { ToolkitError } from '../shared/errors'
 import { AuthUtil, getChatAuthState } from '../codewhisperer/util/authUtil'
 import { telemetry } from '../shared/telemetry/telemetry'
 
+export class OpenApplicationComposerCodeLensProvider implements vscode.CodeLensProvider {
+    public async provideCodeLenses(
+        document: vscode.TextDocument,
+        token: vscode.CancellationToken
+    ): Promise<vscode.CodeLens[]> {
+        const cfnTemplate =
+            CloudFormation.isValidFilename(document.uri) && document.languageId === 'yaml'
```

Comment:
Is there a case where a .yml or .yaml file doesn't have the 'yaml' language ID? This seems redundant to me.

Reply:
I don't know why there would be, I copied this code from the other place I found it used [here](https://github.com/aws/aws-toolkit-vscode/blob/144dfccb9b1936c290586777f1233c11233e526b/src/shared/awsFiletypes.ts#L70). My best guess is that this is trying to avoid the case where the TextDocument is on a windows machine, where the file extension could be changed but the `languageId` may still be `yaml`. If this is the case it seems overly restrictive though so I agree with removing the requirement for the `languageId` 👍